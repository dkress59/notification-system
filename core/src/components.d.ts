import { NotificationType } from "./types";
export namespace Components {
	interface BannerArea {
		/**
		  * Takes the `<banner-notification />`-component's props in camelCase as an argument
		 */
		"spawnBanner": ({ autoHide, autoHideAfterMs, content, headline, type, }: { autoHide?: boolean; autoHideAfterMs?: number; content: string; headline?: string; type?: NotificationType; }) => void;
	}
	interface BannerNotification {
		/**
		  * Whether to automatically hide the banner, or not. If false (or undefined), a dismiss-button will be rendered.
		 */
		"autoHide": boolean;
		/**
		  * The time in milliseconds after which the banner shall be hidden (requires the auto-hide attribute to be set to "true").
		 */
		"autoHideAfterMs": number;
		/**
		  * Entirely dismisses the banner from the DOM
		 */
		"dismiss": () => void;
		/**
		  * If provided, the banner will be rendered with a headline which is styled slightly more prominent than the body text.
		 */
		"headline": string;
		/**
		  * The notification-type of the banner (success | info | warning | error).
		 */
		"type": NotificationType;
	}
	interface ModalNotification {
		/**
		  * If set to 'false' the 'Confirm'-button will be disabled.
		 */
		"condition": boolean | undefined;
		/**
		  * Entirely dismisses the modal from the DOM
		 */
		"dismiss": () => void;
		/**
		  * If provided, the modal will be rendered with a headline which is styled slightly more prominent than the body text.
		 */
		"headline": string;
		/**
		  * Label for the 'Confirm'-button
		 */
		"labelConfirm": string;
		/**
		  * Label for the 'Decline'-button
		 */
		"labelDecline": string;
		/**
		  * Whether to show the 'Confirm'-button.
		 */
		"showConfirm": boolean;
		/**
		  * Whether to show the 'Decline'-button.
		 */
		"showDecline": boolean;
		/**
		  * The notification-type of the modal (success | info | warning | error).
		 */
		"type": NotificationType;
	}
	interface NotificationArea {
		/**
		  * Takes the `<modal-notification />`-component's props in camelCase as an argument
		 */
		"spawnModal": ({ condition, content, headline, showConfirm, showDecline, labelConfirm, labelDecline, type, }: { condition?: boolean; content: string; headline?: string; showConfirm?: boolean; showDecline?: boolean; labelConfirm?: string; labelDecline?: string; type?: NotificationType; }) => void;
		/**
		  * Takes the `<toast-notification />`-component's props in camelCase as an argument
		 */
		"spawnToast": ({ autoHide, autoHideAfterMs, content, headline, type, }: { autoHide?: boolean; autoHideAfterMs?: number; content: string; headline?: string; type?: NotificationType; }) => void;
	}
	interface ToastNotification {
		/**
		  * Whether to automatically hide the toast, or not. If false (or undefined), a dismiss-button will be rendered.
		 */
		"autoHide": boolean;
		/**
		  * The time in milliseconds after which the toast shall be hidden (requires the auto-hide attribute to be set to "true").
		 */
		"autoHideAfterMs": number;
		/**
		  * Entirely dismisses the toast from the DOM
		 */
		"dismiss": () => void;
		/**
		  * If provided, the toast will be rendered with a headline which is styled slightly more prominent than the body text.
		 */
		"headline": string;
		/**
		  * The notification-type of the toast (success | info | warning | error).
		 */
		"type": NotificationType;
	}
}
export namespace Elements {
	interface HTMLBannerAreaElement extends Components.BannerArea, HTMLElement {
	}
	var HTMLBannerAreaElement: {
		prototype: HTMLBannerAreaElement;
		new(): HTMLBannerAreaElement;
	};
	interface HTMLBannerNotificationElement extends Components.BannerNotification, HTMLElement {
	}
	var HTMLBannerNotificationElement: {
		prototype: HTMLBannerNotificationElement;
		new(): HTMLBannerNotificationElement;
	};
	interface HTMLModalNotificationElement extends Components.ModalNotification, HTMLElement {
	}
	var HTMLModalNotificationElement: {
		prototype: HTMLModalNotificationElement;
		new(): HTMLModalNotificationElement;
	};
	interface HTMLNotificationAreaElement extends Components.NotificationArea, HTMLElement {
	}
	var HTMLNotificationAreaElement: {
		prototype: HTMLNotificationAreaElement;
		new(): HTMLNotificationAreaElement;
	};
	interface HTMLToastNotificationElement extends Components.ToastNotification, HTMLElement {
	}
	var HTMLToastNotificationElement: {
		prototype: HTMLToastNotificationElement;
		new(): HTMLToastNotificationElement;
	};
	interface HTMLElementTagNameMap {
		"banner-area": HTMLBannerAreaElement;
		"banner-notification": HTMLBannerNotificationElement;
		"modal-notification": HTMLModalNotificationElement;
		"notification-area": HTMLNotificationAreaElement;
		"toast-notification": HTMLToastNotificationElement;
	}
}
declare namespace LocalJSX {
	interface BannerArea {
	}
	interface BannerNotification {
		/**
		  * Whether to automatically hide the banner, or not. If false (or undefined), a dismiss-button will be rendered.
		 */
		"autoHide"?: boolean;
		/**
		  * The time in milliseconds after which the banner shall be hidden (requires the auto-hide attribute to be set to "true").
		 */
		"autoHideAfterMs"?: number;
		/**
		  * If provided, the banner will be rendered with a headline which is styled slightly more prominent than the body text.
		 */
		"headline"?: string;
		/**
		  * Fires after the elements has transitioned out.
		 */
		"onBannerDismissed"?: (event: CustomEvent<HTMLElement>) => void;
		/**
		  * The notification-type of the banner (success | info | warning | error).
		 */
		"type"?: NotificationType;
	}
	interface ModalNotification {
		/**
		  * If set to 'false' the 'Confirm'-button will be disabled.
		 */
		"condition"?: boolean | undefined;
		/**
		  * If provided, the modal will be rendered with a headline which is styled slightly more prominent than the body text.
		 */
		"headline"?: string;
		/**
		  * Label for the 'Confirm'-button
		 */
		"labelConfirm"?: string;
		/**
		  * Label for the 'Decline'-button
		 */
		"labelDecline"?: string;
		/**
		  * Fires when the 'Confirm'-button is pressed.
		 */
		"onConfirmTriggered"?: (event: CustomEvent<HTMLElement>) => void;
		/**
		  * Fires when the 'Decline'-button is pressed.
		 */
		"onDeclineTriggered"?: (event: CustomEvent<HTMLElement>) => void;
		/**
		  * Fires after the elements has transitioned out.
		 */
		"onModalDismissed"?: (event: CustomEvent<HTMLElement>) => void;
		/**
		  * Whether to show the 'Confirm'-button.
		 */
		"showConfirm"?: boolean;
		/**
		  * Whether to show the 'Decline'-button.
		 */
		"showDecline"?: boolean;
		/**
		  * The notification-type of the modal (success | info | warning | error).
		 */
		"type"?: NotificationType;
	}
	interface NotificationArea {
	}
	interface ToastNotification {
		/**
		  * Whether to automatically hide the toast, or not. If false (or undefined), a dismiss-button will be rendered.
		 */
		"autoHide"?: boolean;
		/**
		  * The time in milliseconds after which the toast shall be hidden (requires the auto-hide attribute to be set to "true").
		 */
		"autoHideAfterMs"?: number;
		/**
		  * If provided, the toast will be rendered with a headline which is styled slightly more prominent than the body text.
		 */
		"headline"?: string;
		/**
		  * Fires after the elements has transitioned out.
		 */
		"onToastDismissed"?: (event: CustomEvent<HTMLElement>) => void;
		/**
		  * The notification-type of the toast (success | info | warning | error).
		 */
		"type"?: NotificationType;
	}
	interface IntrinsicElements {
		"banner-area": BannerArea;
		"banner-notification": BannerNotification;
		"modal-notification": ModalNotification;
		"notification-area": NotificationArea;
		"toast-notification": ToastNotification;
	}
}
