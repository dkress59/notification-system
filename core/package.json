{
  "name": "core",
  "version": "0.0.1",
  "description": "Notification System Web Components",
  "main": "dist/index.cjs.js",
  "files": [
    "dist/bundle.js"
  ],
  "scripts": {
    "build": "webpack --mode production",
    "start": "webpack serve"
  },
  "devDependencies": {
    "autoprefixer": "10.4.7",
    "clean-webpack-plugin": "4.0.0",
    "css-loader": "6.7.1",
    "html-webpack-plugin": "5.5.0",
    "postcss": "8.4.14",
    "postcss-import": "14.1.0",
    "postcss-scss": "4.0.4",
    "sass": "1.52.1",
    "sass-loader": "13.0.0",
    "ts-jest": "28.0.2",
    "ts-loader": "9.3.0",
    "ts-node": "10.7.0",
    "webpack": "5.74.0",
    "webpack-cli": "4.10.0",
    "webpack-dev-server": "4.9.3"
  },
  "license": "MIT",
  "lint-staged": {
    "*.{scss, css}": [
      "yarn fix:style"
    ],
    "*.{js,jsx,ts,tsx}": [
      "yarn fix:script"
    ]
  }
}
