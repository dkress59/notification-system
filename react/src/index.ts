export * from './components/banner-area'
export * from './components/banner-notification'
export * from './components/modal-notification'
export * from './components/notification-area'
export * from './components/toast-notification'
export * from './types'
