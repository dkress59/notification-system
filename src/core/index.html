<!DOCTYPE html>
<html dir="ltr" lang="en">

<head>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, minimum-scale=1.0, maximum-scale=5.0" />
  <title>Notification Web Components</title>

  <script type="module" src="/bundle.js"></script>

  <style>
    * {
      box-sizing: border-box;
    }

    body {
      margin: 0;
      padding: 0;
      font-family: ui-sans-serif, system-ui, -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, "Helvetica Neue", Arial, "Noto Sans", sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji";
    }
  </style>
  <script>
    document.addEventListener('DOMContentLoaded', () => {
      document.querySelectorAll('toast-notification').forEach(toast => {
        toast.addEventListener('toastDismissed', (event) => console.debug(
          'toast dismissed.',
          { event }
        ))
      })
      document.querySelectorAll('modal-notification').forEach(modal => {
        modal.addEventListener('confirmTriggered', (event) => console.debug(
          'modal confirmed.',
          { event }
        ))
        modal.addEventListener('declineTriggered', (event) => console.debug(
          'modal declined.',
          { event }
        ))
      })

      const bannerArea = document.querySelector('banner-area')
      setTimeout(() => {
        bannerArea.spawnBanner({
          content: 'yoyoyo',
          autoHide: true,
        })
      }, 2000)
      setTimeout(() => {
        bannerArea.spawnBanner({
          content: 'yoyoyo',
          type: 'warning',
          headline: 'spawned',
        })
      }, 4000)

      const notificationArea = document.querySelector('notification-area')
      setTimeout(() => {
        notificationArea.spawnModal({
          content: 'yoyoyo',
          type: 'warning',
          headline: 'spawned',
        })
      }, 1500)
      setTimeout(() => {
        notificationArea.spawnModal({
          content: 'yoyoyo',
          headline: 'spawned new',
        })
      }, 4500)
    })
  </script>
</head>

<body>
  <banner-area>
    <banner-notification headline="some headline" type="error">demo content</toast-notification>
  </banner-area>
  <notification-area>
    <toast-notification class="test" headline="some headline" type="info">demo content</toast-notification>
  </notification-area>
</body>

</html>
