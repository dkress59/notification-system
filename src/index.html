<!DOCTYPE html>
<html dir="ltr" lang="en">

<head>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, minimum-scale=1.0, maximum-scale=5.0" />
  <title>Notification Web Components</title>

  <script type="module" src="/build/notification-system.esm.js"></script>
  <script nomodule src="/build/notification-system.js"></script>

  <style>
    * {
      box-sizing: border-box;
    }

    body {
      font-family: sans-serif;
    }

    #notification-container {
      position: fixed;
      z-index: 50;
      top: 0;
      right: 0;
      bottom: 0;
      left: 0;
      pointer-events: none;
      display: flex;
      flex-direction: column;
      align-items: flex-end;
      overflow-y: auto;
    }

    /* #notification-container > * {
        right: 0;
      } */
  </style>
  <script>
    document.addEventListener('DOMContentLoaded', () => {
      document.querySelectorAll('notify-toast').forEach(toast => {
        toast.addEventListener('toastDismissed', (shadowRoot) => console.debug(
          'toast dismissed!',
          { shadowRoot }
        ))
      })
    })
  </script>
</head>

<body>
  <div id="notification-container">
    <!-- <my-component first="Stencil" last="'Don't call me a framework' JS"></my-component>
      <my-component first="more Stencil" middle="please" last="'Don't call me a framework' JS"></my-component> -->
    <notify-toast type="error" headline="Lorem Ipsum">Lorem ipsum dolor sit amet consectetur adipisicing elit, facilis
      necessitatibus sunt quisquam officia quidem.</notify-toast>
    <notify-toast>Mock Content</notify-toast>
    <notify-toast auto-hide>Mock Content</notify-toast>
    <notify-toast type="info" is-hidden>Hidden Content</notify-toast>
    <notify-toast>Mock Content</notify-toast>
  </div>
</body>

</html>