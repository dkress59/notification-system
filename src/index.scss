@import 'css/base.scss';
// @import 'css/util.scss';
