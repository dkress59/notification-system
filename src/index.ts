export { Components, JSX } from './components'
export { BannerArea } from './components/banner-area/banner-area'
export { BannerNotification } from './components/banner-notification/banner-notification'
export { ModalNotification } from './components/modal-notification/modal-notification'
export { NotificationArea } from './components/notification-area/notification-area'
export { ToastNotification } from './components/toast-notification/toast-notification'
