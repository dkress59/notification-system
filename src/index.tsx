import './index.scss'

import React from 'react'
import ReactDOM from 'react-dom'

import { App } from './app'
import { NotificationProvider } from './hooks'

ReactDOM.render(
	<React.StrictMode>
		<NotificationProvider>
			<App />
		</NotificationProvider>
	</React.StrictMode>,
	document.getElementById('app'),
)
