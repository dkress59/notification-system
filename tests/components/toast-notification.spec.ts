import { newSpecPage } from '@stencil/core/testing'

import { ToastNotification } from '../../src'

const mockContent = 'Mock Content'
const mockHeadline = 'Mock Headline'
const mockClassName = 'mock-class-name'

describe('<toast-notification />', () => {
	it('matches snapshot without props', async () => {
		const { root } = await newSpecPage({
			components: [ToastNotification],
			html: '<toast-notification></toast-notification>',
		})
		expect(root).toMatchSnapshot()
	})

	// eslint-disable-next-line jest/no-disabled-tests
	it.skip('matches snapshot with props', async () => {
		const { root } = await newSpecPage({
			components: [ToastNotification],
			html: `<toast-notification class="${mockClassName}" headline="${mockHeadline}">${mockContent}</toast-notification>`,
		})
		expect(root).toMatchSnapshot()
	})
})
