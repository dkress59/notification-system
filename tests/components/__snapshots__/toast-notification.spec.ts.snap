// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<toast-notification /> matches snapshot with props 1`] = `
<toast-notification class="hidden mock-class-name" headline="Mock Headline" style="transition: none;">
  <mock:shadow-root>
    <h4 data-testid="headline">
      Mock Headline
    </h4>
    <span class="icon" data-testid="notify-icon">
      <span>
        ✓
      </span>
    </span>
    <section>
      <slot></slot>
    </section>
    <span class="dismiss" data-testid="dismiss-button">
      ✕
    </span>
  </mock:shadow-root>
  Mock Content
</toast-notification>
`;

exports[`<toast-notification /> matches snapshot without props 1`] = `
<toast-notification class="hidden" style="transition: none;">
  <mock:shadow-root>
    <span class="icon" data-testid="notify-icon">
      <span>
        ✓
      </span>
    </span>
    <section>
      <slot></slot>
    </section>
    <span class="dismiss" data-testid="dismiss-button">
      ✕
    </span>
  </mock:shadow-root>
</toast-notification>
`;
